packages/runtime/datastore/src/channelStorageService.ts

```diff
@@ -27,7 +26,7 @@ export class ChannelStorageService implements IChannelStorageService {
     constructor(
         private readonly tree: ISnapshotTree | undefined,
         private readonly storage: Pick<IDocumentStorageService, "readBlob">,
-        private readonly extraBlobs?: Map<string, string>,
+        private readonly extraBlobs?: Map<string, ArrayBufferLike>,
```

Comment:
there are a few cases where blob is a string originally, should I change all of them to be ArrayBufferLike? @vladsud

Reply:
It depends. I'd assume that existing code passes base64-encoded strings (that's obvious from usage - read() would just return it).
In order to answer you question we need to take a look at usage of this class, i.e. where blobs are coming from. If it's easy to change and if fits into overall flow, then yes, it would be useful to change it to be buffers, as it's consistent with the rest of APIs.
But it's possible that for staging, or some other reasons it might be hard or not natural to pass buffers, and keeping base64-encoded strings might be fine.